gpbft/participant.go

```diff
@@ -155,33 +162,209 @@ func (p *Participant) ValidateMessage(msg *GMessage) (valid ValidatedMessage, er
 		return nil, err
 	}
 
+	// TODO: Refactor validation into its own struct such that it encapsulates
+	//       caching, metrics etc.
+	//       See: https://github.com/filecoin-project/go-f3/issues/561
+
 	var buf bytes.Buffer
-	var marshalled bool
-	err = msg.MarshalCBOR(&buf)
-	if err != nil {
+	var cacheMessage bool
+	if err := msg.MarshalCBOR(&buf); err != nil {
 		log.Warnw("failed to marshal message for cache check", "err", err)
```

Comment:
this is an error

Reply:
As in error log level, or should return error?

If former, I used warn to avoid noise in logs, since caching is not a critical part of validation.
If for whatever reason this error occurs, the user would see about 15k of these per epoch.
Having said that I don't know how likely it is for this error to occur.

If latter, the rationale for proceeding with validation is that whether marshalling succeeds or not we can still validate the message. Just a bit slower and more resource intensive. 

WDYT? Does that make sense?

Reply:
I'd just log with an Error. If this fails, something is really wrong.